src/pages/DelegatoryValidator/Components/TimeDurationIntervalBar.tsx

```diff
@@ -12,37 +14,55 @@ export default function TimeDurationIntervalBar({
     return null;
   }
 
+  const [remainingTime, setRemainingTime] = useState<moment.Duration>();
+  const [percentageComplete, setPercentageComplete] = useState<number>(0);
+
   // the beginning of the unlock cycle
   const startTime = parseTimestamp(timestamp.toString()).subtract(30, "days");
 
   // the end of the unlock cycle
   const unlockTime = parseTimestamp(timestamp.toString());
 
-  // the remaining time of the unlock cycle
-  const now = moment();
-  const remainingTime = moment.duration(
-    unlockTime.valueOf() - now.valueOf(),
-    "milliseconds",
-  );
+  useEffect(() => {
```

Comment:
btw, we do something similar on governance https://github.com/aptos-labs/governance/blob/main/src/pages/Proposal/Header.tsx#L73-L85 not sure if its exactly this use case but might be good to check it out